tests/kernFeatureWriter_test.py

```diff
@@ -1,21 +1,28 @@
 from __future__ import print_function, division, absolute_import, unicode_literals
 
+from textwrap import dedent
+
 import unittest
 from defcon import Font
 
-from ufo2ft.kernFeatureWriter import KernFeatureWriter
+from ufo2ft.featureCompiler import FeatureCompiler
+from ufo2ft.featureWriters import KernFeatureWriter
+
+from fontTools.misc.py23 import SimpleNamespace
 
 
 class KernFeatureWriterTest(unittest.TestCase):
+
     def test_collect_fea_classes(self):
         text = '@MMK_L_v = [v w y];'
         expected = {'@MMK_L_v': ['v', 'w', 'y']}
 
         ufo = Font()
         ufo.features.text = text
-        writer = KernFeatureWriter(ufo)
+        writer = KernFeatureWriter()
+        writer.set_context(ufo)
```

Comment:
This could still be a ctor arg where you call set_context in the ctor, no?

Reply:
No, that is intentional. It is a private method that is called every time the writer.write(ufo) method is called, each time with a different font. The same writer instance configured with some options can be used for multiple fonts.